app/order/workflows.go

```diff
@@ -178,15 +184,31 @@ func (wf *orderImpl) cancelUnavailableFulfillments() {
 }
 
 func (wf *orderImpl) waitForCustomer(ctx workflow.Context) (string, error) {
+	var signal CustomerActionSignal
+
+	s := workflow.NewSelector(ctx)
+
+	timerCtx, cancelTimer := workflow.WithCancel(ctx)
+	t := workflow.NewTimer(timerCtx, customerActionTimeout)
+
+	s.AddFuture(t, func(_ workflow.Future) {
```

Comment:
By not waiting on the future, you'll never know if the workflow cancelled and therefore a workflow cancel will appear as a customer action timeout (but maybe that's what we want, but should mention that then as a comment). For the most part we should encourage users to check their futures for errors.

Reply:
Updated.